### PR TITLE
Minting with metadata

### PR DESCRIPTION
**Pull request by tifrel**

### CODE REVIEW

**Review by sainthiago — APPROVED** [review]

File: packages/sdk/src/mint/mint.ts [hunk 1]
```diff
@@ -5,11 +5,28 @@ import { NearContractCall } from '../execute';
 
 export type MintArgs =  {
   nftContractId?: string;
-  reference: string;
   ownerId: string;
+  metadata: TokenMetadata;
```
Comment (cif) [inline packages/sdk/src/mint/mint.ts]: This is a breaking API change, perhaps we can still accept reference, then make an additional property in `options` called `additionalMetadata` , then in args: `metadata: { reference, ...additionalMetadata }`

Comment (tifrel) [reply packages/sdk/src/mint/mint.ts]: Did we decide not to do any breaking changes? Shouldn't typing + versioning be enough?

Comment (cif) [reply packages/sdk/src/mint/mint.ts]: Fair point, when you're in alpha/beta anything server `0.x` technically all bets are off.  Maybe we release beta this week with the clear message "you should pin versions and look for changes to this API each time you update"

File: packages/sdk/src/mint/mint.ts [hunk 2]
```diff
@@ -26,14 +43,28 @@ export type Splits = Record<string, number>;
...
 
   const { splits, amount, royaltyPercentage } = options;
-  
-  if (nftContractId == null) {
+
+  if (nftContractId === null) {
     throw new Error('You must provide a nftContractId or define a NFT_CONTRACT_ID environment variable to default to');
   }
 
+  if (!noReference && !metadata.reference) {
```
Comment (cif) [inline packages/sdk/src/mint/mint.ts:30]: Very nice

**Review by cif — APPROVED** [review]

File: packages/sdk/src/mint/mint.ts [hunk 3]
```diff
@@ -26,14 +43,28 @@ export type Splits = Record<string, number>;
...
+    throw new Error('You must provide reference in your metadata or explicitly opt out of using reference');
```
Comment (cif) [inline packages/sdk/src/mint/mint.ts]: Looks like the conflict here moved these messages into constants which is nice, be good to follow that pattern as well

Comment (tifrel) [reply packages/sdk/src/mint/mint.ts]: Done

File: packages/sdk/src/mint/mint.ts (hunk 1 above)
Comment (tifrel) [reply packages/sdk/src/mint/mint.ts]: I left it in flexibly now, but I prefer the `metadata.reference` API. Requires some logic to check if it's conflicting or not. If we prioritize API ergonomics and clarity, let's remove it.

Comment (cif) [reply packages/sdk/src/mint/mint.ts]: The current version I pulled still had the breaking change, removed the conflict message and updated the docs.  I think we will just roll the breaking change as part of beta release task today

**Pull request merged**